core/src/test/kotlin/io/bkbn/kompendium/core/util/RouteParsing.kt

```diff
@@ -100,3 +101,28 @@ fun Routing.trailingSlash() {
     }
   }
 }
+
+fun Routing.parameter() {
```

Comment:
can you make this name more descriptive? Reason being is that Kompendium already does support parameters, in the original `this/is/a/{param}` approach.  So maybe call this test `paramWrapper` or something. 

Also... do you happen to know if there is any actual benefit to using this approach over just declaring the route the way I mentioned above?

Additionally, could you add another test to make sure that nested declarations work as expected?

ie 

```
param("a") {
  param("b") {
     // ..
```

Reply:
Good evening, initially I ran into this problem when I tried to install the NotarizedRoute plugin inside a route with a resource.

``` kotlin
@Serializable
@Resource("/test")
class Test(
    val param1: String,
    val param2: String,
    val param3: String
)

routing {
    rosource<Test> {
        install(NotarizedRoute()) {
            ...
        }
        ...
    }
}
```

Reply:
I'm not following... I don't see any `param()` function call in this example

Reply:
In the screenshot below, the resource method from the ktor-server-resources library. There you can see that ParameterRouteSelector or OptionalParameterRouteSelector is added to each query parameter.
![image](https://user-images.githubusercontent.com/60779683/236321082-044592d7-31fe-4b7d-8b91-9aecbc78cf29.png)

Here I have opened the param method, which also adds ParameterRouteSelector.
![image](https://user-images.githubusercontent.com/60779683/236321997-571eb3f3-a972-4ffc-bcd1-25545cf6aa6b.png)

It turns out that the ktor-server-resources library adds query parameters to each resource in the same way as the param method. The param method adds /[{param_name}] to the route path, respectively, and the ktor-server-resources library does the same, because the algorithm for adding query parameters is the same.

If we try to take the route path inside the lambda expression of the resource method, we get /api/test/[param1]/[param2]/[param3]. I observed a similar path in openapi.json, of course, so requests from swagger did not work for me.

I hope I managed to explain.

Reply:
Ah thank you makes sense 👍